tests/test_D2_model_constraints.py

```diff
@@ -188,30 +188,42 @@ def run_parts(margs):
             model = D0.model_building.adding_assets_to_energysystem_model(
                 dict_values, dict_model, model
             )
-            local_energy_system = solph.Model(model)
-            logging.debug("Created oemof model based on created components and busses.")
-            return dict_values, local_energy_system, dict_model
+            return dict_values, model, dict_model
 
         self.dict_values, self.model, self.dict_model = run_parts()
         self.exp_emission_limit = 1000
+        self.exp_min_renewable_share = 0.60
         self.dict_values[CONSTRAINTS].update(
             {MAXIMUM_EMISSIONS: {VALUE: self.exp_emission_limit}}
         )
+        self.dict_values[CONSTRAINTS].update(
+            {MINIMAL_RENEWABLE_FACTOR: {VALUE: self.exp_min_renewable_share}}
+        )
         return
 
+    # This test is to check if the maximum emissions limit as specified by the user has been properly added or not
     def test_constraint_maximum_emissions(self):
+        # Create a solph model using the input values (especially the constraints setup as class variables above)
         model = D2.constraint_maximum_emissions(
-            model=self.model, dict_values=self.dict_values
+            model=solph.Model(self.model), dict_values=self.dict_values,
         )
+        # Assert that the constraint value within the model is equal to the emission factor set above as a class variable
+        # This test will fail if the constraint value for maximum emissions limit provided by the user is not properly added to the oemof model of the energy system
```

Comment:
This is actually described below: partly in the assertion message and partly just reading the content of `assert()`. I would at least delete the second line. Please adapt first line, in case you want to keep it.

Reply:
I will delete these lines. I added them for my own understanding when going through code :D